### PR TITLE
feat: AIACA nodes in CA chain

### PR DESCRIPTION
**Pull request by JonasBK**
## Description

Include AIACA nodes in CA chain (paths of IssuedSignedBy and EnterpriseCAFor edges)

## Motivation and Context

This PR addresses: BED-4454

The same CA is often represented by multiple AD objects (RootCA, EnterpriseCA, and AIACA nodes). An intermediate CA exist as an EnterpriseCA and an AIACA by default, but only one of them is required for a given CA chain to be valid. This PR will ensure BloodHound detects valid CA chains in the case there is no EnterpriseCA node for an intermediate CA. 

Additionally, EnterpriseCA nodes will now get an EnterpriseCAFor edge to the corresponding AIACA node if their cert thumbprint is the same i.e. it is the same CA. That will enable BloodHound to create additional CA chains across forests.

## How Has This Been Tested?

Modified the harness test for IssuedSignedBy and created one for EnterpriseCAFor.

Manually checked the composition with test data.

## Screenshots (optional):

image

image

## Types of changes

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

File: packages/go/analysis/ad/esc3.go [hunk 1]
```diff
@@ -688,8 +688,8 @@ func ADCSESC3Path6_7Pattern(ecaId graph.ID, domainId graph.ID) traversal.Pattern
 			query.KindIn(query.Relationship(), ad.IssuedSignedBy, ad.EnterpriseCAFor),
 		)).
 		OutboundWithDepth(0, 0, query.And(
-			query.Kind(query.Start(), ad.EnterpriseCA),
 			query.KindIn(query.Relationship(), ad.IssuedSignedBy, ad.EnterpriseCAFor),
+			query.KindIn(query.End(), ad.EnterpriseCA, ad.AIACA, ad.RootCA),
```
Comment (urangel) [inline packages/go/analysis/ad/esc3.go:692]: This pattern looks slightly different from the other places this has been updated. Are these blocks equivalent? 

```go
		OutboundWithDepth(0, 0, query.And(
			query.KindIn(query.Relationship(), ad.IssuedSignedBy, ad.EnterpriseCAFor),
			query.KindIn(query.End(), ad.EnterpriseCA, ad.AIACA, ad.RootCA),
		)).
		Outbound(query.And(
			query.KindIn(query.Relationship(), ad.RootCAFor),
			query.Equals(query.EndID(), domainId),
		))
```

```go
              OutboundWithDepth(0, 0, query.And(
			query.KindIn(query.Relationship(), ad.IssuedSignedBy, ad.EnterpriseCAFor),
			query.KindIn(query.End(), ad.EnterpriseCA, ad.AIACA),
			)).	
		Outbound(
			query.And(
				query.KindIn(query.Relationship(), ad.IssuedSignedBy, ad.EnterpriseCAFor),
				query.Kind(query.End(), ad.RootCA),
			)).
		Outbound(
			query.And(
				query.KindIn(query.Relationship(), ad.RootCAFor),
				query.Equals(query.EndID(), domainID),
			))
```

If they are equivalent, I would recommend writing them the same way but if they are not equivalent maybe add a comment as to how and why it is done differently here.

File: cmd/api/src/test/integration/harnesses.go [hunk 2]
```diff
@@ -1753,18 +1758,70 @@ func (s *IssuedSignedByHarness) Setup(graphTestContext *GraphTestContext) {
...
+	s.AIACA1_2 = graphTestContext.NewActiveDirectoryAIACAWithThumbprint("aiaca1_2", sid, "b")
+	s.AIACA2_1 = graphTestContext.NewActiveDirectoryAIACAWithThumbprint("aiaca2_1", sid, "c")
+
+	s.RootCA1.Properties.Set(ad.CertChain.String(), []string{"a"})
+	s.RootCA2.Properties.Set(ad.CertChain.String(), []string{"b"})
+	s.EnterpriseCA1.Properties.Set(ad.CertChain.String(), []string{"a"})
+	s.EnterpriseCA2.Properties.Set(ad.CertChain.String(), []string{"c", "a"})
+	s.AIACA1_1.Properties.Set(ad.CertChain.String(), []string{"a"})
+	s.AIACA1_2.Properties.Set(ad.CertChain.String(), []string{"b"})
+	s.AIACA2_1.Properties.Set(ad.CertChain.String(), []string{"c", "a"})
```
[the remark this answers is not in the document]
Comment (JonasBK) [reply cmd/api/src/test/integration/harnesses.go:1816]: Good point 👍

File: packages/go/analysis/ad/esc3.go (hunk 1 above)
Comment (JonasBK) [reply packages/go/analysis/ad/esc3.go:692]: Good point 👍

**Comment by JonasBK** [comment]
Thanks for the comments @urangel! Your suggestions are now implemented

**Review by urangel — APPROVED** [review]
Nice work Jonas

**Pull request merged**